src/CFlatToWasm.ml

```diff
@@ -981,16 +1001,14 @@ let mk_func env { args; locals; body; name; ret; _ } =
     in
     (if name <> "WasmSupport_align_64" then Debug.mk env debug_enter else []) @
     read_highwater @
+    [ dummy_phrase (W.Ast.LocalSet (mk_var (env.n_args + 4))) ] @
```

Comment:
here, instead of leaving the top-of-the-(memory)-stack pointer on top of the (operand) stack, we save it in the reserved local